### PR TITLE
Fix all cases where entities would crash when creating or spawning

### PR DESCRIPTION
**Pull request by simon816**
This fixes crashes that occur when trying to create an entity, or spawn an entity directly after creation.

### CODE REVIEW

File: src/main/java/org/spongepowered/common/mixin/core/world/MixinWorld.java [hunk 1]
```diff
@@ -277,28 +288,41 @@ public boolean isBlockPowered(int x, int y, int z) {
...
+            return Optional.absent();
+        }
+
+        net.minecraft.world.World world = (net.minecraft.world.World) (Object) this;
+
+        // Not all entities have a single World parameter as their constructor
+        if (entityClass.isAssignableFrom(EntityLightningBolt.class)) {
+            entity = (Entity) new EntityLightningBolt(world, position.getX(), position.getY(), position.getZ());
+        } else if (entityClass.isAssignableFrom(EntityEnderPearl.class)) {
+            EntityArmorStand tempEntity = new EntityArmorStand(world, position.getX(), position.getY(), position.getZ());
```
Comment (Zidane) [inline src/main/java/org/spongepowered/common/mixin/core/world/MixinWorld.java]: While hilarious, unsure if we want an armorstand to be the projectile source of an EnderPearl when spawned this way :P.

Comment (simon816) [inline src/main/java/org/spongepowered/common/mixin/core/world/MixinWorld.java]: I meant to unset this after the constructor and have an `UnknownProjectileSource` as the default. I'll fix this now.

File: src/main/java/org/spongepowered/common/mixin/core/world/MixinWorld.java [hunk 2]
```diff
@@ -277,28 +294,71 @@ public boolean isBlockPowered(int x, int y, int z) {
...
+            return Optional.absent();
+        }
+
+        net.minecraft.world.World world = (net.minecraft.world.World) (Object) this;
+
+        // Not all entities have a single World parameter as their constructor
+        if (entityClass.isAssignableFrom(EntityLightningBolt.class)) {
+            entity = (Entity) new EntityLightningBolt(world, x, y, z);
+        } else if (entityClass.isAssignableFrom(EntityEnderPearl.class)) {
+            EntityArmorStand tempEntity = new EntityArmorStand(world, x, y, z);
```
Comment (Zidane) [inline src/main/java/org/spongepowered/common/mixin/core/world/MixinWorld.java]: Still meant for this to be EntityArmorStand?

Comment (simon816) [inline src/main/java/org/spongepowered/common/mixin/core/world/MixinWorld.java]: Yes, the constructor must take an `EntityLivingBase` and I think that an armor stand is the cheapest object to construct.

**Pull request closed without merging**